File: Part 2 - Regression/Section 8 - Decision Tree Regression/decision_tree_regression .R
#Decision Tree Regression 

#Importing the dataset
dataset= read.csv('Position_Salaries.CSV')
dataset=dataset[2:3]

#splitting the dataset into the Training set and Test set 
#install.packages('caTools')
#library(caTools)
#set.seed(123)
#split = sample.split(dataset$Purchased,SplitRatio = 0.8)
#training_set=subset(dataset,split==TRUE)
#test_set=subset(dataset,split==FALSE)

#Feature Scaling 
# training_set[, 2:3] = scale(training_set[, 2:3])
# test_set[, 2:3]=scale(test_set[, 2:3])

#Fitting Decision Tree Regression to tht dataset 
regressor =  rpart(formula = Salary ~ .,
                   data = dataset,
                   control = rpart.control(minsplit = 1))

#Predict a new result 
y_pred =  predict(regressor,data.frame(Level = 6.5))

#visualising Decision Tree Regression  result 
library(ggplot2)
ggplot()+
  geom_point(aes(x = dataset$Level ,y =dataset$Salary),
             colour = 'red')+
  geom_line(aes(x = dataset$Level, y = predict(regressor, newdata = dataset)),
            colour = 'blue')+
  ggtitle('Truth or Bluff(Decision Tree Regression)')+
  xlab('Level')+
  ylab('Salary')

#visualising the Regression  Model result (for higher resolution and smoother curve)
library(ggplot2)
x_grid = seq(min(dataset$Level),max(dataset$Level),0.01)
ggplot()+
  geom_point(aes(x = dataset$Level ,y =dataset$Salary),
             colour = 'red')+
  geom_line(aes(x = x_grid, y = predict(regressor, newdata = data.frame(Level = x_grid))),
            colour = 'blue')+
  ggtitle('Truth or Bluff(Decision Tree Regression)')+
  xlab('Level')+
  ylab('Salary')

File: Part 2 - Regression/Section 8 - Decision Tree Regression/decision_tree_regression .py
# Decision Tree Regression 
"""
Created on Wed Mar 22 00:43:18 2017

@author: ali5c_000
"""
# Importing the libraries 
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd 
#Importing the dataset

dataset= pd.read_csv('Position_Salaries.csv')
X= dataset.iloc[:,1:2].values
y=dataset.iloc[:,2].values

#splitting the dataset into the Training set and Test set 
"""from sklearn.cross_validation import train_test_split
X_train, X_test, Y_train, Y_test= train_test_split(X,Y, test_size = 0.2, random_state=0)"""

#Feature Scaling 
"""from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
X_train = sc_X.fit_transform(X_train)
X_test = sc_X.transform(X_test)"""
              
#Fitting Decision tree Regression to the Dataset
from sklearn.tree import DecisionTreeRegressor
regressor = DecisionTreeRegressor(random_state = 0)
regressor.fit(X,y)

#Predicting result 
y_pred = regressor.predict(6.5)

#Visualing the Decision tree Regression result (for higher resolution and smother curve)
X_grid = np.arange(min(X),max(X),0.01)
X_grid = X_grid.reshape(len(X_grid),1)
plt.scatter(X,y, color =  'red')
plt.plot(X_grid,regressor.predict(X_grid), color = 'blue')
plt.title('Truth or Bluff (Regression Model)')
plt.xlabel('Position level')
plt.ylabel('Salary')
plt.show()